pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -246,12 +246,12 @@ func (d *Driver) Start() error {
 		return err
 	}
 
-	// virtio-vsock device
-	dev, err = config.VirtioVsockNew(d.DaemonVsockPort, d.VsockPath, true)
+	netDev, err := config.VirtioNetNew("5a:94:ef:e4:0c:ee")
```

Comment:
This duplicates a value from `cmd/crc/cmd/daemon.go`, this at least deserves a comment, probably best to add it to `d`.

Reply:
I'd add the `UnixgramSocketPath` to `driver` to be similar to `DaemonVsockPort` and `VsockPath`.
Maybe we can even support both vsock and unixgram modes in `driver_darwin.go` depending on which values are set in `d`?

Reply:
I have added code to set vsock mode only if `UnixgramSockPath` is not set but I'm setting the path here:
https://github.com/crc-org/crc/pull/4682/files#diff-bd1e0081558a01b12d1bfbb4eedac6696b0f7bc6571e2637d6207a9f7779e38bR70
so it won't be empty